terraform/python/ec2/asg/main.tf

```diff
@@ -0,0 +1,231 @@
+    name   = "vpc-id"
+    values = [aws_default_vpc.default.id]
+  }
+}
+
+resource "tls_private_key" "ssh_key" {
+  algorithm = "RSA"
+  rsa_bits = 4096
+}
+
+resource "aws_key_pair" "aws_ssh_key" {
+  key_name = "instance_key-${var.test_id}"
+  public_key = tls_private_key.ssh_key.public_key_openssh
+}
+
+locals {
+  ssh_key_name        = aws_key_pair.aws_ssh_key.key_name
+  private_key_content = tls_private_key.ssh_key.private_key_pem
+}
+
+data "aws_ami" "ami" {
+  owners = ["amazon"]
+  most_recent      = true
+  filter {
+    name = "name"
+    values = ["al20*-ami-minimal-*-x86_64"]
+  }
+  filter {
+    name   = "state"
+    values = ["available"]
+  }
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+  filter {
+    name   = "image-type"
+    values = ["machine"]
+  }
+
+  filter {
+    name   = "root-device-name"
+    values = ["/dev/xvda"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+}
+
+resource "aws_launch_configuration" "launch_configuration" {
+  image_id      = data.aws_ami.ami.id
+  instance_type = "t3.micro"
+  key_name = local.ssh_key_name
+  associate_public_ip_address = true
+  iam_instance_profile = "APP_SIGNALS_EC2_TEST_ROLE"
+  security_groups = [aws_default_vpc.default.default_security_group_id]
+
+  user_data = <<-EOF
+    #!/bin/bash
+
+    # Install Python and wget
+    sudo yum install wget -y
+    sudo yum install unzip -y
+    sudo dnf install -y python3.9
+    sudo dnf install -y python3.9-pip
+
+    # Copy in CW Agent configuration
+    agent_config='${replace(replace(file("../../../ec2/amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'
+    echo $agent_config > amazon-cloudwatch-agent.json
+
+    # Get and run CW agent rpm
+    ${var.get_cw_agent_rpm_command}
+    sudo rpm -U ./cw-agent.rpm
+    sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json
+
+    # Get ADOT Wheel and install it
+    ${var.get_adot_wheel_command}
+
+    # Get and run the sample application with configuration
+    aws s3 cp ${var.sample_app_zip} ./python-sample-app.zip
+    unzip -o python-sample-app.zip
+
+    # Export environment variables for instrumentation
+    cd ./django_frontend_service
+    # Delete the requests requirement as it is installed already using rpm
+    sudo sed -i '/requests/d' ./requirements.txt
+    python3.9 -m pip install -r requirements.txt
+    export DJANGO_SETTINGS_MODULE="django_frontend_service.settings"
+    export OTEL_PYTHON_DISTRO="aws_distro"
+    export OTEL_PYTHON_CONFIGURATOR="aws_configurator"
+    export OTEL_METRICS_EXPORTER=none
+    export OTEL_TRACES_EXPORTER=otlp
+    export OTEL_AWS_APP_SIGNALS_ENABLED=true
+    export OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT=http://localhost:4315
+    export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4315
```

Comment:
Have we double checked this is correct?

Reply:
Changed

Reply:
The new endpoints are supposed to be 
```
OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT=http://localhost:4316/v1/metrics
OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4316/v1/traces
```

But they do not work in this version currently. Reverted back to 4315 for now